### DISCUSSION TITLE
Getting back into using Hydrogen (silent drumkits)

### DISCUSSION THREAD

**Discussion by LHud** [hydrogen-music/hydrogen#1555] [Hydrogen on Windows]
Hi all,
I had been using Ableton for drums for quite some time but there are just some elements of Hydrogen that make drum creation so much better.  To that end I've installed the latest version, I think, 1.1.1-c845af2f, and have rounded up all of my drumkits I've created or downloaded over the years into my drumkit folder but only one of them provides sound.  They work where they were installed before, my older desktop machine using Hydrogen version 1.0 I believe, but I'm using a new laptop now and they load, at least the name of the kit and the filenames of all the instruments, but there is no samples attached to them.  The drumkits have all of the sounds intact and there are .xml files in the drumkit folders as well.  Is there a way to easily create the .h2drumkit file from these folders or do I have to go through the painstakingly slow method of re-creating the kits with all of the sound files one by one?  I would think there was, I even tried exporting one of the silent kits into a new folder which did create the .h2drumkit file but it didn't provide any sound.  I am on a Windows 10 machine with a Scarlet 2i2 as my in/out sound source.  It isn't my hardware, the factory kits and one of my downloaded kits work fine, all the rest are dead quiet.  Thanks ahead for any help/suggestions.

ps, this is probably a frequent request from time to time, honestly I do try to find this information for myself but I'm finding that a lot of the time it is difficult to search for something specific in most forums so hoping this request doesn't aggravate...

**Comment by theGreatWhiteShark** [comment]
Hey @LHud,

> Is there a way to easily create the .h2drumkit file from these folders or do I have to go through the painstakingly slow method of re-creating the kits with all of the sound files one by one?

It sounds like a small bug. We'll fix it together and there will hopefully no need to resort to manual editing of the drumkit files. ;)

Could you start the program via console using the following command and paste the output in here?

```
"C:\Program Files\Hydrogen\hydrogen.exe" -V Debug
```

During the startup Hydrogen attempts to load the general structure of all drumkits (but not the samples) and might tell what is going wrong. Could you also attach one of the drumkits that is not working properly? (possibly a small one)

**Reply by LHud** [reply]
Hi,
Sorry about the delay, I work A LOT of hours daily.  I ran the -v debug from the command line but it didn't relay any different info inside Hydrogen.  There is a blue banner topside that states the following, I think it does this normally too, without the command line but by starting Hydrogen normally:

**Song Drumkit Samples
Some samples used in this song could not be loaded.  This may be because it uses an older default drumkit.  This might be fixed by opening a new drumkit.**

There is a button inside the banner that says Open drumkit.

I've attached a drumkit that does not work.  This drumkit is listed in the data/drumkits folder and is also visible inside the program.  Strangely all of my kits are visible inside the program, working or not, but not all of them are listed inside the drumkit folder.  I'm thinking that is a clue somehow to why I'm experiencing problems with them.

[Ltibbits_RockKit.zip](https://github.com/hydrogen-music/hydrogen/files/8547619/Ltibbits_RockKit.zip)

Thanks for your help!

**Reply by theGreatWhiteShark** [reply]
> Sorry about the delay,

Don't worry. Some here.

> I've attached a drumkit that does not work. This drumkit is listed in the data/drumkits folder and is also visible inside the program. Strangely all of my kits are visible inside the program, working or not, but not all of them are listed inside the drumkit folder. I'm thinking that is a clue somehow to why I'm experiencing problems with them.
>
> [Ltibbits_RockKit.zip](https://github.com/hydrogen-music/hydrogen/files/8547619/Ltibbits_RockKit.zip)

First of all, the drumkit you posted works perfectly fine on my end.

The drumkit loading is still somewhat strange within Hydrogen. When adding a sample to a layer you can load it from an arbitrary. However, if you save the resulting drumkit and load it again - e.g. while reopening the song - Hydrogen expects the sample to be located in the corresponding kit directory within drumkit folder. All drumkits must be located there or they won't work. Also, if the sample was unable to be loaded, it will be removed from when storing the song.

Is it still not working after renaming the kit directory in the drumkit folder and importing the .h2drumkit you attached via the main menu? Maybe moving the kits to the drumkit folder will already fix some of them.

> I ran the -v debug from the command line but it didn't relay any different info inside Hydrogen. There is a blue banner topside that states the following, I think it does this normally too, without the command line but by starting Hydrogen normally:

Ah, sorry. You are working with a release version. Could you try the command with one of the following (more recent) versions?

Windows 64 bit: https://ci.appveyor.com/api/buildjobs/vra0kocu0extpiu2/artifacts/Hydrogen-1.1.1-win64.exe
Windows 32 bit: https://ci.appveyor.com/api/buildjobs/yi72mj0bggaofsgc/artifacts/Hydrogen-1.1.1-win32.exe
